ironfish/src/wallet/scanner/walletScanner.ts

```diff
@@ -308,20 +312,40 @@ export class WalletScanner {
    */
   private async refreshScanningAccounts(): Promise<void> {
     this.scanningAccounts = (await this.getScanningAccountsWithHead()).map(
-      ({ account, head }) => ({ account, scanFrom: head }),
+      ({ account, head }) => {
+        const scanFrom = head || {
+          sequence: (account.createdAt?.sequence ?? GENESIS_BLOCK_SEQUENCE) - 1,
+        }
+        return { account, scanFrom }
+      },
     )
   }
 
-  private getEarliestHead(): HeadValue | null {
-    let earliestHead = null
-    for (const { scanFrom: head } of this.scanningAccounts) {
-      if (!head) {
-        return null
+  private async getEarliestHead(): Promise<HeadValue | null | 'none'> {
```

Comment:
Does `'none'` mean that all of the `scanFrom`s are ahead of the chain head?

Reply:
Would it work to return the chain head in that case instead of the special `'none'` value?

Reply:
I think its both
1. earliest is ahead of the chain
2. there are no scanning accounts

Reply:
I'm not sure if returning the head would work for this case where we break https://github.com/iron-fish/ironfish/pull/5652/files#diff-4a43e4a9c484bea01d7248d51bc40948e7bbdd13c5bef3fdbfe0a02c7210f85dR138. If we set chain processor to head of the chain then maybe that's not the same as the chain processor's head (because it was set previously)

Idk its a good thought. I think I'll keep it as none for now though so don't have to think through / refactor that break statement